networkx/generators/tests/test_sudoku_graph.py

```diff
@@ -0,0 +1,57 @@
+"""Unit tests for the :mod:`networkx.generators.sudoku_graph` module."""
+
+import pytest
+import networkx as nx
+
+
+def test_sudoku_negative():
+    """Verify that an error is raised when generating a Sudoku graph of order -1."""
+    pytest.raises(nx.NetworkXError, nx.sudoku_graph, n=-1)
+
+
```

Comment:
I think it’s possible to make the tests a bit more concise by parameterizing them using pytest.

Here’s a strawman skeleton:

```python
@pytest.parametrize(“n”, [0, 1, 2, 3])
def test_sudoku_generator(n):
    G = nx.sudoku_graph(n)
    assert not G.is_directed()
    assert not G.is_multigraph()

    assert G.number_of_nodes() == n**4
    # not thinking straight right now, but probably putting in formula for number of edges is better
```

Since we’re doing a formal review here, I would highly recommend using this parametrized testing paradigm, rather than a series of example-based tests, because it then opens up the door to using property-based tests later (if NetworkX ever adopts [Hypothesis] for testing). 

[Hypothesis]: http://hypothesis.readthedocs.io/en/latest

@dschult and @jarrodmillman, would you be willing to chime in here?

Reply:
Thanks! I have parameterized the tests.